src/Middleware/RequestThrottling/src/RequestThrottlingMiddleware.cs

```diff
@@ -16,9 +15,11 @@ namespace Microsoft.AspNetCore.RequestThrottling
     /// </summary>
     public class RequestThrottlingMiddleware
     {
-        private readonly RequestQueue _requestQueue;
+        private readonly SemaphoreSlim _queueSemaphore;
```

Comment:
Why isn't this just a counter with a lock? I see the call _queueSemaphore.WaitAsync(), but it's never actually awaited. You just reject the request if the returned Task is incomplete, but you could do the exact same thing much more simply with an integer and lock object.